### PR TITLE
chore: update all minor and patch dependencies

### PR DESCRIPTION
**Pull request by XhmikosR**
Also

* switch to lockfile v2
* update to `@types/jest@27` to be in sync with the jest version

I know you have renovate installed but it seems those PRs are stuck. Either way, I didn't update any major deps, but synchronized the deps between package.json and the lock file.

I'd like to remove make-dir hence why I made this patch. If you prefer to go with the renovate PRs, that's fine as long as we get things moving soon :)

### CODE REVIEW

**Comment by XhmikosR** [comment]
1163 -> 981 packages :)

**Comment by XhmikosR** [comment]
@erezrokah LGTY now? I still think it's better to use lockfile v2 going forward and this slightly flattens the deps tree.

Either way this and #214 can probably land in a minor/patch version bump :)

**Comment by XhmikosR** [comment]
@erezrokah I can't keep rebasing this, either merge it or close it.

I still think it's a good change, but it's your project, your rules. 🙂

**Comment by XhmikosR** [comment]
I reverted the package.json changes since you wanted that, although it's good to synchronize them from time to time especially in major releases.

Anyway, you can either merge or close, this doesn't make any difference to consumers as it is right now.

**Review by erezrokah — APPROVED** [review]
@XhmikosR thanks and sorry for the late reply. I wanted to release the PRs one by one to have small releases.

I kept the lock file upgrade to v2, however since we use ranges in our `package.json` I don't think that change is necessary (consumers will get the latest patch/minor versions when they install this library).
Bumping the versions in `package.json` will force this lib to use the latest versions, but can also prevent de-duping of packages.

TLDR: lets leave the ranges permissive, and let consumers decide.

**Comment by erezrokah** [comment]
> I reverted the package.json changes

spider-man
